src/types.ts

```diff
@@ -215,21 +215,30 @@ export interface IAnonymousMetadataPartial {
   platform_user_id: string;
 }
 
+export interface IAttachmentsFolder {
+  id: string;
+  readWriteToken?: string;
+}
+
 /**
  * cf. IRunTimeMetadataBase, from
  * https://github.com/concord-consortium/lara/blob/master/lara-typescript/src/interactive-api-client/metadata-types.ts#L47
  * and partial export code at
  * https://github.com/concord-consortium/lara/blob/c40304a14ef495acdf4f9fd09ea892c7cc98247b/app/models/interactive_run_state.rb#L110
  */
 export interface IExportableAnswerMetadataBase {
-  question_id: string;
+  question_id: string;    // converted from refId (e.g. "managed_interactive_404")
   question_type: string;
-  id: string;
+  id: string;             // randomly generated id (e.g. uuid)
   type: string;
   answer_text?: string;
   answer?: any;
   submitted: boolean | null;
   report_state: string;
+  attachmentsFolder?: IAttachmentsFolder;
+  // TODO: do we want to store the public url? aren't we transitioning to expiring urls?
+  // should this be stored in some non-persistent place so it'll be looked up next time?
+  attachments?: Record<string, string>;  // name/key => url
```

Comment:
I was thinking the URL here is the unsigned URL to the object in S3. So when a read operation happens, the client would get the credentials (if it doesn't already have them) from the token service. And then sign this unsigned URL.
The url would be something like:
`https://interactive-attachments.s3.aws.com/attachments/[token-service-record-id]/[session-id]/[name]`

The first part of that URL: 
`https://interactive-attachments.s3.aws.com/attachments/[token-service-record-id]` could be taken from the info retrieved from the token-service. 

So you could just store the remaining path: `[session-id]/[name]`.   Or you could just store the `[session-id]` and then compute the path from that.

Not storing the full url does seem useful. This way we could actually change the attachments bucket, or folder and we wouldn't have to worry  updating all of the interactive state metadata.

Reply:
The value I'm passing in is generated at https://github.com/concord-consortium/activity-player/blob/df7fe8dcc1e439c819829b478c549955e915a6d9/src/portal-api.ts#L412, which seemed like it might be appropriate.

Reply:
We're going around in circles a bit. As you've/we've said before, the sessionId is insufficient to access an attachment written by a collaborator. I'm now storing the folder id and the publicPath.